plugins/guests/darwin/cap/mount_vmware_shared_folder.rb

```diff
@@ -81,9 +81,15 @@ def self.mount_vmware_shared_folder(machine, name, guestpath, options)
                   content = @apply_firmlinks[machine.id][:content].join("\n")
                   # Write out the synthetic file
                   comm.sudo("echo -e #{content.inspect} > /etc/synthetic.conf")
-                  if @apply_firmlinks[:bootstrap]
+                  if @apply_firmlinks[machine.id][:bootstrap]
+                    case machine.guest.capability("flavor")
+                    when :big_sur
+                      apfs_bootstrap_flag = "-t"
+                    else
+                      apfs_bootstrap_flag = "-B"
+                    end
                     # Re-bootstrap the root container to pick up firmlink updates
-                    comm.sudo("/System/Library/Filesystems/apfs.fs/Contents/Resources/apfs.util -B")
+                    comm.sudo("/System/Library/Filesystems/apfs.fs/Contents/Resources/apfs.util #{apfs_bootstrap_flag}", good_exit: [0, 253])
```

Comment:
if `apfs.util -t` has returns `253` it seems to be successful?

Reply:
I think it would be good to set the valid exit values in the conditionals as well, since with `-B` we know `0` is expected but with `-t` it could be `253` as well.